From f0aa34156f8d06b14d065c99368b4f541fd7c955 Mon Sep 17 00:00:00 2001
From: Lin Fuzhen <fuzhen.lin@freescale.com>
Date: Wed, 6 Mar 2013 18:10:21 +0800
Subject: [PATCH 170/602] ENGR00243620 imx5x/imx6x: Enable adb security
 feature

Enable adb security feature, JB4.2.2 add public key authentification
when connect the PC host via adb  , this feature will  keep adb  offline
until device and PC pass the RSA autherntification. SO it also requests
updating the adb tool in PC host side to latest SDK(at least JB4.2.2).

Signed-off-by: Lin Fuzhen <fuzhen.lin@freescale.com>
---
 imx5x/init.rc    | 3 +++
 imx6/etc/init.rc | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/imx5x/init.rc b/imx5x/init.rc
index 4af7897..4f44c91 100644
--- a/imx5x/init.rc
+++ b/imx5x/init.rc
@@ -396,6 +396,9 @@ on boot
 # Set this property so surfaceflinger is not started by system_init
     setprop system_init.startsurfaceflinger 0
 
+#  Enable adb security for JB4.2.2
+    setprop ro.adb.secure 1
+
 # Default Read Ahead value for sdcards
     write /sys/block/mmcblk0/queue/read_ahead_kb 2048
     write /sys/block/mmcblk1/queue/read_ahead_kb 2048
diff --git a/imx6/etc/init.rc b/imx6/etc/init.rc
index 45d7c2f..392d01f 100755
--- a/imx6/etc/init.rc
+++ b/imx6/etc/init.rc
@@ -391,6 +391,9 @@ on boot
 # Set this property so surfaceflinger is not started by system_init
     setprop system_init.startsurfaceflinger 0
 
+#  Enable adb security for JB4.2.2
+    setprop ro.adb.secure 1
+
 # set this to not let jni check crash, just warnning message.
     setprop dalvik.vm.jniopts warnonly
 
-- 
1.8.0

